theano/gpuarray/opt.py

```diff
@@ -600,58 +905,90 @@ def local_gpuajoin_1(node):
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Split])
-def local_gpua_split(node, context_name):
-    return GpuSplit(node.op.len_splits)
+@register_opt2([tensor.Split], 'fast_compile')
+def local_gpua_split(op, context_name, inputs, outputs):
+    # TODO use props
+    return GpuSplit(op.len_splits)
 
 
 @register_opt('fast_compile')
 @op_lifter([tensor.Subtensor])
-def local_gpua_subtensor(node, context_name):
-    x = node.inputs[0]
+def local_gpua_subtensor(op, context_name, inputs, outputs):
+    x = inputs[0]
     if (x.owner and isinstance(x.owner.op, HostFromGpu)):
         gpu_x = x.owner.inputs[0]
         if (gpu_x.owner and
                 isinstance(gpu_x.owner.op, GpuFromHost) and
                 # And it is a shared var or an input of the graph.
                 not gpu_x.owner.inputs[0].owner):
-            if len(x.clients) == 1:
-                if any([n == 'output' or any([isinstance(v.type, GpuArrayType)
-                                              for v in n.inputs + n.outputs])
-                        for n, _ in node.outputs[0].clients]):
-                    return
-                else:
-                    return [host_from_gpu(gpu_x.owner.op(node.outputs[0]))]
-
-    return GpuSubtensor(node.op.idx_list)
+            if len(x.clients) == 1 and len(outputs[0].clients) == 1:
+                return
```

Comment:
Sure. My point is that this function (with the old decorator) is not exactly reverted yet.